hydromt/data_adapter/dataframe.py

```diff
@@ -177,7 +178,15 @@ def to_file(
 
         """
         kwargs.pop("bbox", None)
-        obj = self.get_data(time_tuple=time_tuple, variables=variables, logger=logger)
+        obj = self.get_data(
+            time_tuple=time_tuple,
+            variables=variables,
+            handle_nodata=handle_nodata,
+            logger=logger,
+        )
+
+        if obj is None:
```

Comment:
In data_catalog,export_data, the output fn_out, etc are explicitly asked and passed. If you return None here, it needs to be handled in that function I think:

 https://github.com/Deltares/hydromt/blob/3ef7af4d4a6c0fc24f0aaa08dc5d5def2d2d8227/hydromt/data_catalog.py#L1225-L1241